lib/AST/Builtins.cpp

```diff
@@ -1958,6 +2023,14 @@ ValueDecl *swift::getBuiltinValueDecl(ASTContext &Context, Identifier Id) {
     return getAutoDiffPopFromTape(Context, Id);
   case BuiltinValueKind::AutoDiffDestroyTape:
     return getAutoDiffDestroyTape(Context, Id);
+  case BuiltinValueKind::AutoDiffGetJVP:
+    return getAutoDiffGetAssociatedFunction(Context, Id,
+                                            AutoDiffAssociatedFunctionKind::JVP,
+                                            /*order*/ 1, /*arity*/ 1);
+  case BuiltinValueKind::AutoDiffGetVJP:
+    return getAutoDiffGetAssociatedFunction(Context, Id,
+                                            AutoDiffAssociatedFunctionKind::VJP,
+                                            /*order*/ 1, /*arity*/ 1);
```

Comment:
As you said, there's a mismatch between `getAutoDiffGetAssociatedFunction` (which supports arbitrary order and arity) and `Builtin.autodiffGet(JVP|VJP)` (which supports order 1 and arity 1).

Could you please expand on how you plan to handle this mismatch? Is the only solution to create many versions of the builtins like `Builtin.autodiffGetJVP_Arity2_Order2_Throwing`, as you noted in the PR description?

Reply:
`getAutoDiffGetAssociatedFunction` is a builder, and `Builtin.autodiffGet(JVP|VJP)` is a concrete pair of functions created by that builder. When we overload, we will simply declare more builtins, which will then call the builder with a different arity/order/throwing-ness. I think this can also be simplified by extending the parser to recognize the suffixes.